### PR TITLE
Sync Updates - Partitions

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-16513): [Concept] Partitions
- (DOCSP-16521): [Reference/Explain] Partition Strategies

### Staged Changes (Requires MongoDB Corp SSO)

- [Partitions](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/sync-updates/sync/partitions/)
- [Partition Strategies](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/sync-updates/sync/partition-strategies/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sync/partitions.txt [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+A **partition** represents a subset of the documents in a synced cluster that
+all share a logical access pattern and have the same read/write permissions for
```
Comment (mongomoe) [inline source/sync/partitions.txt:19]: what is a "logical access pattern"? 

Can we possibly replace that fragment with more intuitive wording?

File: source/sync/partitions.txt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+.. _partition-key:
+
+Partition Key
+~~~~~~~~~~~~~
+
+The **partition key** is the name of the field that Realm uses to determine
+which partition a given document in a synced MongoDB cluster belongs to. Your
+app's partition key is a central part of the sync data model that you should
+consider during development. Once you have chosen a partition key and
+:ref:`enabled sync <enable-sync>`, you cannot reassign the partition key to a
```
Comment (mongomoe) [inline source/sync/partitions.txt:56]: "... you cannot reassign the partition key"
Is this true? Or should it be "...you cannot reassign the partition key to a different field without terminating and reenabling sync"?

File: source/sync/partitions.txt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+could be either:
+
+- A property that already exists in each document and that logically partitions
+  data. For example, if every document is private to a specific user you might
+  indicate the user's ID value in an ``owner_id`` field and then use that as the
+  partition key.
+
+- A synthetic property that you create specifically to partition data (e.g.
+  ``_partition``). This is useful for more complex data models that don't
+  include a natural partition key or that require granular access permissions.
```
Comment (mongomoe) [inline source/sync/partitions.txt:69]: 59-69 is a great explanation 💯 🥇

File: source/sync/partitions.txt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+A partition's value is the identifier for its corresponding synced realm. To
+work with the data in a given partition from a client application, you specify
+the partition's value when you open its corresponding synced realm. If you
+change a partition value in {+atlas-short+}, Realm interprets the change as a
+delete from the old partition and an insert into the new partition.
+
+.. warning::
+
+   If you change a document's partition value, you will lose any un-synced
```
Comment (mongomoe) [inline source/sync/partitions.txt:157]: nitpick: lose any un-synced -> lose any updates that you have not synced

File: source/sync/partition-strategies.txt [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+--------
+
+A :ref:`partition strategy <partition-strategy>` is an approach to configuring
+how your app partitions data in order to handle a particular use case. This page
+describes common partition strategies that you can adapt and combine to fit your
+application.
+
+Realm uses :ref:`partitions <sync-partitions>` to map objects between synced
+realms on client devices and documents stored in MongoDB Atlas. Partitions are a
+core part of your synced data model that determine each user's read and write
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:25]:
```suggestion
core part of your synced data model that determines each user's read and write
```

File: source/sync/partition-strategies.txt [hunk 6]
```diff
@@ -0,0 +1,493 @@
...
+   strategy.
+
+.. _partition-strategy-firehose:
+
+Firehose Strategy
+-----------------
+
+A firehose partition strategy groups all documents into a single partition so
+that every user syncs all of your app's data to their device. This approach is
+functionally a decision to not partition data and is suitable only for basic
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:62]:
```suggestion
functionally a decision not to partition data and is suitable only for basic
```

File: source/sync/partition-strategies.txt [hunk 7]
```diff
@@ -0,0 +1,493 @@
...
+        { teams: ["Southside Rockets", "Uptown Bombers"], score: { ... }, date: ... }
+      ]
+      
+      collection teams: [
+        { name: "Brook Ridge Miners" }
+        { name: "Southside Rockets" }
+        { name: "Uptown Bombers" }
+      ]
+   
+   The total number of historical games is relatively small and a small number
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:103]:
```suggestion
   The total number of historical games is relatively small, and a small number
```

File: source/sync/partition-strategies.txt [hunk 8]
```diff
@@ -0,0 +1,493 @@
...
+   of local teams only play a few games each year, so most devices should be
+   able to download all of the game data for easy offline access. In this case,
+   the firehose strategy is appropriate and the documents don't need to have a
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:106]:
```suggestion
   the firehose strategy is appropriate, and the documents don't need to have a
```

File: source/sync/partition-strategies.txt [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+        Team { name: "Southside Rockets" }
+        Team { name: "Uptown Bombers" }
+      ]
+
+.. _partition-strategy-user:
+
+User Strategy
+-------------
+
+A user partition strategy groups private documents that belong to a user into a
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:130]: Should it be: 

"A user partition strategy is used to group private documents ..."

or maybe "You can use a user partition strategy to group private documents ..."

File: source/sync/partition-strategies.txt [hunk 10]
```diff
@@ -0,0 +1,493 @@
...
+partition specific to that user. This approach is effective when each document
+has a specific owner, often identified by a username or ID, and no other users
+need access to the data.
+
+- **Data Security:** Data in a user partition is specific to a given user and
+  may contain information private to that user. In general each user syncs only
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:136]:
```suggestion
  may contain information private to that user. In general, each user syncs only
```

File: source/sync/partition-strategies.txt [hunk 11]
```diff
@@ -0,0 +1,493 @@
...
+  their own partition and other users cannot access documents in the partition.
+
+- **Storage Capacity:** In general, each user only syncs data from their own
+  partition so their data must fit within their device's storage constraints.
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:140]:
```suggestion
  partition, so their data must fit within their device's storage constraints.
```

File: source/sync/partition-strategies.txt (hunk 9 above)
Comment (mongomoe) [reply source/sync/partition-strategies.txt:130]: same for the other strategy intro sentences as well ^

File: source/sync/partition-strategies.txt [hunk 12]
```diff
@@ -0,0 +1,493 @@
...
+
+A channel partition strategy groups documents relevant to a common topic or
+domain into a partition specific to that area. Users can choose to access or
+subscribe to specific channels, often identified by a name or ID, from a public
+list.
+
+- **Data Security:** Data in a channel partition is specific to a given topic or
+  area that a user can choose to access. You can limit a user's access to
+  specific channels or prevent them from accessing channels entirely, but if a
+  user has read or write permission for a channel then they can access any
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:300]:
```suggestion
  user has read or write permission for a channel, then they can access any
```

File: source/sync/partition-strategies.txt [hunk 13]
```diff
@@ -0,0 +1,493 @@
...
+
+- **Storage Capacity:** Users can choose to sync data from any allowed channel.
+  The combined data from all of the channels a given user accesses must fit
+  within their device's storage constraints. You can use this strategy to
+  partition public or semi-private data sets that would otherwise not fit within
+  storage constraints.
+
+.. example::
+   
+   A social app lets users create chatrooms based around specific topics. Users
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:311]:
```suggestion
   A social app lets users create chatrooms based on specific topics. Users
```

File: source/sync/partition-strategies.txt [hunk 14]
```diff
@@ -0,0 +1,493 @@
...
+A region partition strategy groups documents that are relevant to a specific
+location or region into partitions specific to those areas.
+
+- **Data Security:** Data in a region partition is specific to a given
+  geographic area. You can limit a user's access to just the region they're
+  currently in or give access to data on a region-by-region basis.
+
+- **Storage Capacity:** The storage implications of a region strategy depend on
+  the size and usage patterns for each region. Users typically only sync data in
+  their own region, so at a minimum the data for any given region should fit
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:389]:
```suggestion
  their own region, so at a minimum, the data for any given region should fit
```

File: source/sync/partition-strategies.txt [hunk 15]
```diff
@@ -0,0 +1,493 @@
...
+  within a device's storage constraints. If users sync multiple regions you may
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:390]:
```suggestion
  within a device's storage constraints. If users sync multiple regions, you may
```

File: source/sync/partition-strategies.txt [hunk 16]
```diff
@@ -0,0 +1,493 @@
...
+   key as an optional field not present in any document. Realm maps any document
+   that doesn't include a partition value to the *null partition*.
+
+.. example::
+   
+   A local sports app lets users download and browse game scores and statistics
+   for historical high school baseball games. Consider the following documents
+   in the ``games`` and ``teams`` collections:
+   
+   .. code-block:: javascript
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:86]: should there be commas each object in the array for the following codeblock?

File: source/sync/partition-strategies.txt [hunk 17]
```diff
@@ -0,0 +1,493 @@
...
+        Playlist { name: "Deep Focus", owner_id: "PUBLIC", song_ids: [ ... ] }
+      ]
+
+.. _partition-strategy-team:
+
+Team Strategy
+-------------
+
+A team partition strategy groups private documents shared by a team of users
+(e.g. a store location's emplyee, a band's members, etc) into a partition
```
Comment (mongomoe) [inline source/sync/partition-strategies.txt:204]: typo:
```suggestion
(e.g. a store location's employee, a band's members, etc) into a partition
```

File: source/sync/partitions.txt [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+Consider the following when choosing a partition key:
+
+- Partition keys must have one of the following property types: ``String``,
+  ``ObjectID``, ``Int``, ``Long``
+
+- {+service-short+} clients should never modify the partition value directly.
+  Any field that clients can modify cannot be used as a partition key.
+
+- Partition keys use the same field name in every synced document. The partition
+  key should not collide with an another field name in any object's model.
```
Comment (lindseymoore) [inline source/sync/partitions.txt:85]: 'any other' instead of 'an another'

File: source/sync/partition-strategies.txt (hunk 16 above)
Comment (nlarew) [reply source/sync/partition-strategies.txt:86]: it's not actually JS so commas aren't needed - this is the format that we were using previously

**Pull request merged**